centreon/src/Core/UserProfile/Infrastructure/Repository/DbReadUserProfileRepository.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\UserProfile\Infrastructure\Repository;
+
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Infrastructure\DatabaseConnection;
+use Core\Common\Infrastructure\Repository\AbstractRepositoryRDB;
+use Core\Common\Infrastructure\Repository\SqlMultipleBindTrait;
+use Core\UserProfile\Application\Repository\ReadUserProfileRepositoryInterface;
+use Core\UserProfile\Domain\Model\UserProfile;
+
+/**
+ * @phpstan-type _UserProfileRecord array{
+ *    id:int,
+ *    contact_id:int,
+ *    favorite_dashboards:string|null
+ * }
+ */
+final class DbReadUserProfileRepository extends AbstractRepositoryRDB implements ReadUserProfileRepositoryInterface
```

Comment:
In this class, you must to catch PDO exceptions and throw a RepositoryException. When you catch PDO Exceptions you must to log with a custom message and a context please. Your interface must to throw a RepositoryException for each methods to implement. Thanks.

Reply:
done ! to valid with this PR : #6241 @adr-mo